src/coreComponents/fileIO/Table/TableData.hpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableData.hpp
+ */
+
+#ifndef GEOS_COMMON_TableData_HPP
+#define GEOS_COMMON_TableData_HPP
+
+#include "common/DataTypes.hpp"
+#include "common/Format.hpp"
+
+namespace geos
+{
+
+/**
+ * @brief Class for managing table data
+ */
+class TableData
+{
+public:
+
+  /**
+   * @brief Add a row to the table.
+   * The values passed to addRow (can be any type).
+   * @param args Cell values to be added to the row.
+   */
+  template< typename ... Args >
+  void addRow( Args const & ... args );
+
+  /**
+   * @brief Add a row to the table
+   * @param row A vector of string representing a row
+   */
+  void addRow( std::vector< string > const & row );
+
+  /**
+   * @brief Reset data in the table
+   */
+  void clear();
+
+  /**
+   * @return The rows of the table
+   */
+  std::vector< std::vector< string > > const & getTableDataRows() const;
+
+  /**
+   * @brief Get all error messages
+   * @return The vector of error messages
+   */
+  std::vector< string > const & getErrorMsgs() const;
+
+  /**
+   * @brief Set an error message
+   * @param msg The error msg to vector
+   */
+  void addErrorMsgs( string const & msg );
+
+protected:
```

Comment:
Who's inheriting from `TableData`? If no one, then consider putting this `private`.
If you think that some day, you'll set it back to `protected`, then we can do it afterwards.